ee/codegen/src/project.ts

```diff
@@ -349,168 +354,170 @@ ${errors.slice(0, 3).map((err) => {
     return { inputs, workflow, nodes };
   }
 
-  private generateNodes(
+  private async generateNodes(
     nodeIds: string[]
-  ): BaseNode<WorkflowDataNode, BaseNodeContext<WorkflowDataNode>>[] {
+  ): Promise<BaseNode<WorkflowDataNode, BaseNodeContext<WorkflowDataNode>>[]> {
     const nodes: BaseNode<
       WorkflowDataNode,
       BaseNodeContext<WorkflowDataNode>
     >[] = [];
 
-    nodeIds.forEach(async (nodeId) => {
-      let node: BaseNode<WorkflowDataNode, BaseNodeContext<WorkflowDataNode>>;
-
-      const nodeContext = this.workflowContext.getNodeContext(nodeId);
-      const nodeData = nodeContext.nodeData;
-
-      const nodeType = nodeData.type;
-      switch (nodeType) {
-        case WorkflowNodeTypeEnum.SEARCH: {
-          node = new SearchNode({
-            workflowContext: this.workflowContext,
-            nodeContext: nodeContext as TextSearchNodeContext,
-          });
-          break;
-        }
-        case WorkflowNodeTypeEnum.SUBWORKFLOW: {
-          const variant = nodeData.data.variant;
-          switch (variant) {
-            case "INLINE":
-              node = new InlineSubworkflowNode({
-                workflowContext: this.workflowContext,
-                nodeContext: nodeContext as InlineSubworkflowNodeContext,
-              });
-              break;
-            case "DEPLOYMENT":
-              node = new SubworkflowDeploymentNode({
-                workflowContext: this.workflowContext,
-                nodeContext: nodeContext as SubworkflowDeploymentNodeContext,
-              });
-              break;
-            default: {
-              assertUnreachable(variant);
+    await Promise.all(
+      nodeIds.map(async (nodeId) => {
```

Comment:
This forEach instead of Promise.all(array.map()) was causing all sorts of problems trying to catch errors